webknossos/script-collection/globalize_floodfill.py

```diff
@@ -0,0 +1,364 @@
+            ), "volume annotation must have file_len=1"
+            assert (
+                volume_data_wkw.header.voxel_type
+                == fallback_layer_wkw.header.voxel_type
+            ), "Volume annotation must have same dtype as fallback layer"
+            with wkw.Dataset.open(
+                str(output_path),
+                wkw.Header(
+                    voxel_type=fallback_layer_wkw.header.voxel_type,
+                    file_len=1,
+                    block_type=wkw.Header.BLOCK_TYPE_LZ4HC,
+                ),
+            ) as merged_wkw:
+                annotation_bucket_paths = list_bucket_paths(volume_data_path)
+
+                annotated_count = 0
+                bucket_boxes = list(
+                    BoundingBox.from_wkw_dict(bbox)
+                    .align_with_mag(Mag(BUCKET_SIZE), ceil=True)
+                    .chunk(BUCKET_SHAPE)
+                )
+                total_bucket_count = len(bucket_boxes)
+                for count, bucket_bbox in enumerate(bucket_boxes):
+                    if count % 100 == 0:
+                        print(f"Processing bucket {count} of {total_bucket_count}...")
+                    if (
+                        bucket_path_for_pos(bucket_bbox.topleft)
+                        in annotation_bucket_paths
+                    ):
+                        data = volume_data_wkw.read(
+                            bucket_bbox.topleft.to_np(), BUCKET_SHAPE
+                        )
+                        merged_wkw.write(bucket_bbox.topleft.to_np(), data)
+                        annotated_count += 1
+                    else:
+                        data = fallback_layer_wkw.read(
+                            bucket_bbox.topleft.to_np(), BUCKET_SHAPE
+                        )
+                        merged_wkw.write(bucket_bbox.topleft.to_np(), data)
+
+    print(
+        f"Combined {annotated_count} volume-annotated buckets with"
+        f" {total_bucket_count - annotated_count} from fallback layer."
+    )
+
+
+def list_bucket_paths(volume_data_layer_path: Path) -> List[Path]:
+    return [
+        wkw_path.relative_to(volume_data_layer_path)
+        for wkw_path in volume_data_layer_path.rglob("*/*/*.wkw")
+    ]
+
+
+def bucket_path_for_pos(position: Vec3Int) -> Path:
+    x_bucket, y_bucket, z_bucket = position // BUCKET_SIZE
+    return Path(f"z{z_bucket}/y{y_bucket}/x{x_bucket}.wkw")
+
+
+def main(args: Namespace) -> None:
+    nml_regex = re.compile(
+        r'<userBoundingBox .*name="Limits of flood-fill \(source_id=(\d+), target_id=(\d+), seed=([\d,]+), timestamp=(\d+)\)".*topLeftX="(\d+)" topLeftY="(\d+)" topLeftZ="(\d+)" width="(\d+)" height="(\d+)" depth="(\d+)" />'
+    )
+
+    bboxes: List[FloodFillBbox] = []
+    nml_file = open(args.nml_path, "r", encoding="utf-8")
+    lines = nml_file.readlines()
+    nml_file.close()
+    for line in lines:
+        matches = nml_regex.findall(line)
+        for match in matches:
```

Comment:
Please mention https://github.com/scalableminds/webknossos-libs/issues/482 as a todo to clean this up. The existing skeleton API should make reading this way easier, but since the name is not exposed, yet, it's not really possible.